zebra/connected.c

```diff
@@ -263,11 +266,33 @@ void connected_up(struct interface *ifp, struct connected *ifc)
 	if (zrouter.asic_offloaded)
 		flags |= ZEBRA_FLAG_OFFLOADED;
 
-	rib_add(afi, SAFI_UNICAST, zvrf->vrf->vrf_id, ZEBRA_ROUTE_CONNECT, 0,
-		flags, &p, NULL, &nh, 0, zvrf->table_id, metric, 0, 0, 0);
+	/*
+	 * It's possible to add the same network and mask
+	 * to an interface over and over.  This would
+	 * result in an equivalent number of connected
+	 * routes.  Just add one connected route in
+	 * for all the addresses on an interface that
+	 * resolve to the same network and mask
+	 */
+	for (ALL_LIST_ELEMENTS_RO(ifp->connected, cnode, c)) {
+		struct prefix cp;
 
-	rib_add(afi, SAFI_MULTICAST, zvrf->vrf->vrf_id, ZEBRA_ROUTE_CONNECT, 0,
-		flags, &p, NULL, &nh, 0, zvrf->table_id, metric, 0, 0, 0);
+		PREFIX_COPY(&cp, CONNECTED_PREFIX(c));
+		apply_mask(&cp);
+
+		if (prefix_same(&cp, &p))
```

Comment:
I think you can break this for loop the moment count becomes more than 1 and may be even return from here. 

if we didn't add any route, Do we need schedule  LSP processing ? If not, that should also be covered with this if check.

Same comment for delete work flow.

Reply:
fixed